lib/bson/parser/deserializer.js

```diff
@@ -27,17 +28,18 @@ var deserialize = function(buffer, options, isArray) {
     throw new Error(`bson size must be >= 5, is ${size}`);
   }
 
-  if (options.allowObjectSmallerThanBufferSize && buffer.length < size) {
-    throw new Error(`buffer length ${buffer.length} must be >= bson size ${size}`);
+  if (options.allowObjectSmallerThanBufferSize && Buffer.byteLength(buffer) < size) {
```

Comment:
Why `Buffer.byteLength`?

Reply:
`length` is not a real property on `Buffer`. It comes from the underlying Uint8Array which is not always consistent depending on the environment. `byteLength` will always produce the correct result which will be the same as `length`.